File: src/db-backend/test-programs/python/python_hcr_flow_test/mymodule_v2.py
# mymodule.py — version 2 (post-reload)

def compute(n):
    """v2: triples the input."""
    return n * 3

def transform(value, n):
    """v2: multiplies value by n."""
    return value * n

def aggregate(history):
    """v2: sums the history list (same behavior)."""
    return sum(history)

File: src/db-backend/test-programs/ruby/ruby_hcr_flow_test/mymodule_v2.rb
# mymodule.rb -- version 2 (post-reload)

def compute(n)
  # v2: triples the input
  n * 3
end

def transform(value, n)
  # v2: multiplies value by n
  value * n
end

def aggregate(history)
  # v2: sums the history array (same behavior)
  history.sum
end
